### PR TITLE
refactor(derek): revamp resources pattern

### PR DESCRIPTION
**Pull request by amberRrucker**
RSWEB-8629

Refactoring and revamping the resources pattern.
I've made the pattern maintainable and chosen maintainable classes for the icons to avoid conflicts.
The EBPs should still just be able to choose the same classes, we can just fill them into the middle of the class.

This is viewable on any page that has resources on it but I used this one:
http://localhost:3000/derek/templates/product/product-overview

Please review in mobile and with the hover state as well.

### CODE REVIEW

**Comment by runrog** [comment]
@amberRrucker  just a few things to document:
1) we plan on switching some of the icons to the ones used in the designs (waiting on font additions)
2) the same goes for the caret icon. we want to use the long arrow right.

**Comment by amberRrucker** [comment]
Just to note for code review the new icons will be in the new font icon refresh but wont be available until the EBPs update the pages with icons they wanted to get rid of so we can merge that.

File: styleguide/derek/pattern-components/additional-resources/_partials/additional-resource.ejs [hunk 1]
```diff
@@ -13,15 +14,20 @@ PDF: fa-file-pdf-o
 API/Code: fa-file-code-o
 Support/KC: fa-wrench
 -->
-<div class="additional-resource">
+<div class="resourceWrapper resource-<%- type -%>Icon">
   <a href="javascript:void(0)">
```
Comment (bighappyface) [inline styleguide/derek/pattern-components/additional-resources/_partials/additional-resource.ejs]: Let's apply a `track-click` class to the anchor to remind the us that tracking on this pattern is considered.

Comment (amberRrucker) [inline styleguide/derek/pattern-components/additional-resources/_partials/additional-resource.ejs]: you got it. excellent idea.

**Comment by brantdarilek** [comment]
LGTM
👍 on dev

**Comment by runrog** [comment]
@amberRrucker what happened to the hover state / margin bottom. I'm looking at the latest...

**Comment by WidgetsBurritos** [comment]
[run ci]

File: styleguide/derek/pattern-components/resourceCenter-resource/_partials/resourceCenter-resource.ejs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<!-- Default description to allow for partial override to demonstrate really long text and equal height -->
+<% if(!type) { var type = 'pdf'; } %>
+<% if(!title) { var title = 'Resource Title'; } %>
+<% if(!desc) { var desc = 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean m'; } %>
+<!-- Additional Resources Pattern -->
+<!-- Icon Options -->
+<!--
+Case Study: fa-suitcase
+Github: fa-github-square
```
Comment (bighappyface) [inline styleguide/derek/pattern-components/resourceCenter-resource/_partials/resourceCenter-resource.ejs]: Are these applicable?

**Pull request merged**